extensions/amp-carousel/0.1/slides.js

```diff
@@ -49,7 +49,7 @@ export class AmpSlides extends BaseCarousel {
     this.slides_.forEach((slide, i) => {
       this.setAsOwner(slide);
       // Only the first element is initially visible.
-      slide.style.display = i > 0 ? 'none' : 'block';
```

Comment:
Yeah the problem with `display: none` is that resources will totally ignore the element and won't schedule preload for it although the carousel is asking the resources to schedule preload for them. Some more context on #2742 

If we expect a big hit on performance maybe a compromise is to use both `visibility` and `display`. Where `visibility` would be used for for +1/-1 window length (e.g. for slider this is 1 slide before and 1 slide after) and `display:none` the rest outside that window. This would complicate the logic slightly but if we expect the performance to be worth it maybe it's worth getting it right. 

In anyway, since we might be working on big changes for carousel maybe we can keep that in mind in the new design.